poetry.lock

```diff
@@ -119,7 +119,7 @@ python-versions = ">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*"
 
 [[package]]
 name = "coverage"
-version = "6.3"
+version = "6.3.1"
```

Comment:
You probably ran a simple `poetry lock` after you've removed the `typing-extensions`, next time use `poetry lock --no-update` to avoid needlessly updating the dependencies and only add/remove dependencies that are no longer necessary.  Afterall, the goal of this PR is purely the removal of `typing-extensions`, not an update of the packages we use. This should probably be reverted (it's not a huge issue to update, but it just doesn't really belong here).

Note that running `poetry lock --no-update` will probably not do any changes since as I mentioned in my general comment about this PR above, some of our runtime dependencies require `typing-extensions` on their runtime anyway, so the dependency will still be there, it just won't be explicitly listed in our runtime deps.